cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/spark/SparkAppUsingFileSet.java

```diff
@@ -105,20 +107,48 @@ public void run(SparkContext context) {
       String input = context.getRuntimeArguments().get("input");
       String output = context.getRuntimeArguments().get("output");
 
-      // read the dataset
-      JavaPairRDD<Long, String> inputData = context.readFromDataset(input, Long.class, String.class);
-
-      // create a new RDD with the same key but with a new value which is the length of the string
-      JavaPairRDD<String, Integer> stringLengths = inputData.mapToPair(
-        new PairFunction<Tuple2<Long, String>, String, Integer>() {
-          @Override
-          public Tuple2<String, Integer> call(Tuple2<Long, String> pair) throws Exception {
-            return new Tuple2<>(pair._2(), pair._2().length());
-          }
-        });
-
-      // write the character count to dataset
-      context.writeToDataset(stringLengths, output, String.class, Integer.class);
+      String inputPartitionKey = context.getRuntimeArguments().get("inputKey");
+      String outputPartitinKey = context.getRuntimeArguments().get("outputKey");
```

Comment:
Also, from the code below, it seems like it is not representing a key, but rather outputPartitionTime?